src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,95 @@
-import { useState } from 'react';
+import { useEffect, useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (newMovie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [newTitle, setNewTitle] = useState('');
+  const [newDescription, setNewDescription] = useState('');
+  const [newImgUrl, setNewImgUrl] = useState('');
+  const [newImdbUrl, setNewImdbUrl] = useState('');
+  const [newImdbId, setNewImdbId] = useState('');
+  const [isFormValid, setIsFormValid] = useState(false);
+
+  const handleReset = () => {
+    setNewTitle('');
+    setNewDescription('');
+    setNewImgUrl('');
+    setNewImdbUrl('');
+    setNewImdbId('');
+  };
+
+  const handleFormSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    const newMovie: Movie = {
+      title: newTitle,
+      description: newDescription,
+      imgUrl: newImgUrl,
+      imdbUrl: newImdbUrl,
+      imdbId: newImdbId,
+    };
+
+    onAdd(newMovie);
+    handleReset();
+  };
+
+  useEffect(() => {
+    setIsFormValid(
+      !!newTitle.trim()
+        && !!newImgUrl.trim()
+        && !!newImdbUrl.trim()
+        && !!newImdbId.trim(),
+    );
+  }, [newTitle, newImgUrl, newImdbUrl, newImdbId]);
```

Comment:
it's bad way to call useEffect each time when your fields was changes
```suggestion
  const isFormValid = 
      !!newTitle.trim()
        && !!newImgUrl.trim()
        && !!newImdbUrl.trim()
        && !!newImdbId.trim()
```